core/src/com/unciv/logic/civilization/CapitalConnectionsFinder.kt

```diff
@@ -92,9 +92,10 @@ class CapitalConnectionsFinder(private val civInfo: CivilizationInfo) {
         if (cityToConnectFrom.wasPreviouslyReached(transportType, overridingTransportType))
             return
 
-        val bfs = BFS(cityToConnectFrom.getCenterTile()) { tileFilter(it) }
+        val bfs = BFS(cityToConnectFrom.getCenterTile()) {
+              it.getOwner()?.let { owner -> civInfo.hasOpenBordersTo(owner) } ?: true && (it.isCityCenter() || tileFilter(it)) }
```

Comment:
`{ -> Boolean? } ?: true` I'd write as `!= false`, matter of taste. But `?.let` should always be checked for alternatives with better readability...
```kotlin
        val bfs = BFS(cityToConnectFrom.getCenterTile()) {
            val owner = it.getOwner() ?: return@BFS tileFilter(it)
            civInfo.hasOpenBordersTo(owner) && (it.isCityCenter() || tileFilter(it))
        }
```
should be equivalent as long as the assumption that tiles with isCityCenter() true always have an owner holds.

Reply:
Now that I look at this line again I'm also not 100% convinced that Kotlin interprets this as
```(getOwner()?.let { [...] } ?: true) && (it.isCityCenter() || tileFilter(it))```
rather than
```getOwner()?.let { [...] } ?: (true && (it.isCityCenter() || tileFilter(it)))```
which would very likely not be what you meant.

Reply:
[Presley _has_ a higher precedence than ampamp](https://kotlinlang.org/docs/reference/grammar.html#expressions)

Reply:
Initially, I wrote exactly what SomeTroglodyte suggests: two lines with a temp variable.
However, it seemed to me too complex: more text - more efforts to read. Less code - less space for errors there.
Also, mentioning `tileFilter(it)` twice in the same condition smells like "copy-paste coding", what if there will be a need of one more condition to add? Copying it same as tileFilter? No, thanks, we already had `it.isCityCenter()` in 3 different places.
So I simplified that in one line with the same logic we use when readying: does it has an owner? `?.let` if yes, read further, if no, jump to `?:` and continue reading.